specs/docs/merkle-tree/merkle-tree.md

```diff
@@ -0,0 +1,84 @@
+# Merkle Tree
+
+To keep our merkle tree verification succinct, efficient and friendly with [SnarkyJS](https://github.com/o1-labs/snarkyjs). A poseidon merkle tree will be used to prove the immutability of the data. The **Sparse Merkle Tree** is utilized as an adaptation of the conventional Merkle Tree.
+
+## Sparse Merkle Tree (SMT)
+A Sparse Merkle Tree (SMT) is a variant of a standard Merkle tree that is optimized for scenarios where the data set is very large, but only a small portion of it is populated with values. You could refer to the following article to learn more: [What’s a Sparse Merkle Tree?](https://medium.com/@kelvinfichter/whats-a-sparse-merkle-tree-acda70aeb837).
+
+### Advantages of SMT
+Sparse Merkle Trees (SMTs) offer several benefits:
+
+1) **Efficiency**: They allow efficient storage of large, sparse datasets with minimal memory overhead.
+2) **Security**: Like traditional Merkle trees, they provide tamper-proof data structure and enable cryptographic verification of data integrity.
```

Comment:
Does Sparse Merkle Tree have the same security issue with Merkle Tree?

Reply:
Yes, it does. It means if any unauthorized changes are made, they will be detected in the root.

Reply:
When I mentioned security issue, I meant Sparse Merkle Tree have the same Merkle Tree vulnerability that would lead to a false proof.